### PR TITLE
fix: [TECH-663] restructure/clean new enrollment event page

### PR DESCRIPTION
**Pull request by JoakimSM**
- Clean unnecessary code (code from the single event page that is not needed in the enrollment event page)
- Replace code that gets data from currentSelections
- Restructure the Widget folder and remove complexity from the `DataEntry.component.js` file. The restructuring was really pragmatic, hence you will see code that should have been refactored, but I had to skip it at this point.
- Since we have made a Widget that resides in the components folder it should be reusable in the app. Made the necessary changes.

### CODE REVIEW

File: src/core_modules/capture-core/components/WidgetEnrollmentEventNew/Validated/useLifecycle.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    useEffect(() => {
+        if (orgUnit) {
+            dispatch(batchActions([
+                ...getOpenDataEntryActions(program, formFoundation, orgUnit, dataEntryId, itemId),
+            ]));
+        }
+    }, [dispatch, program, formFoundation, orgUnit, dataEntryId, itemId]);
+
+    // temporary logic until we clean up the data from the redux store
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WidgetEnrollmentEventNew/Validated/useLifecycle.js:27]: Hey @JoakimSM, 
Can you create a Jira ticket so we don't forget to clean this temporary logic? 
Thanks!

File: src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.component.js [hunk 2]
```diff
@@ -17,9 +17,9 @@ const styles = ({ typography }) => ({
 });
 
 const EnrollmentAddEventPagePain = ({
-    programStage,
     classes,
-}) => (
+    ...passOnProps // needs destructering when introducing scope selector
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.component.js:21]: 👌

File: src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js [hunk 3]
```diff
@@ -1,28 +1,60 @@
...
+    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
+
+    // TODO: Validate query params
     const { program } = useProgramInfo(programId);
     const programStage = [...program.stages.values()].find(item => item.id === stageId);
     if (!programStage) {
         return <span>[program stage placeholder]</span>;
     }
+
+    // TODO: Get data from enrollment collection for the rules engine
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js:45]: Is this todo related to https://jira.dhis2.org/browse/TECH-635?

File: src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js [hunk 4]
```diff
@@ -1,28 +1,60 @@
...
+
+    const handleCancel = useCallback(() => {
+        dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
+    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
+
+    const handleSave = useCallback(() => {
+        dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
+    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
+
+    // TODO: Validate query params
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js:38]: Hey, 
Do we have a Jira ticket to address the query params (`programId`, `stageId`, `orgUnitId`, `teiId`, `enrollmentId`) validation? 
Thanks!

File: src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js (hunk 3 above)
Comment (JoakimSM) [reply src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js:46]: Yes, it is :+1. Ticket-number added to comment.

File: src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js (hunk 4 above)
Comment (JoakimSM) [reply src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js:38]: Thanks. Ticket created just now and comment updated.

File: src/core_modules/capture-core/components/WidgetEnrollmentEventNew/Validated/useLifecycle.js (hunk 1 above)
Comment (JoakimSM) [reply src/core_modules/capture-core/components/WidgetEnrollmentEventNew/Validated/useLifecycle.js]: Thanks for the reminder. Ticket created.

File: src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js [hunk 5]
```diff
@@ -1,28 +1,60 @@
...
+            teiId: query.teiId,
+            enrollmentId: query.enrollmentId,
         }),
         shallowEqual,
     );
+
+    const dispatch = useDispatch();
+
+    const handleCancel = useCallback(() => {
+        dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
```
[the remark this answers is not in the document]
Comment (JoakimSM) [reply src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js:31]: Hey, it's actually implemented already. It's dealt with before this callback is triggered, in the 'withCancelButton.js' HOC.

File: src/core_modules/capture-core/components/WidgetEnrollmentEventNew/Validated/Validated.component.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        display: 'flex',
+        alignItems: 'center',
+        padding: spacersNum.dp8,
+    },
+    icon: {
+        paddingRight: spacersNum.dp8,
+    },
+});
+
+const ValidatedPlain = ({
```
[the remark this answers is not in the document]
Comment (JoakimSM) [reply src/core_modules/capture-core/components/WidgetEnrollmentEventNew/Validated/Validated.component.js:23]: I'm definitely open to name suggestions for this one 🙂  The point is that when you get here, the input has been validated and your access rights has been verified. I know I've been calling similar components 'Main' in the past, but I tried to make it a bit more descriptive. Anyway, a better naming suggestion here is welcome.

**Review by simonadomnisoru — APPROVED** [review]

**Pull request merged**